brainfuck/brainfuck.d

```diff
@@ -22,11 +24,14 @@ final:
 
 class Program {
   string code;
```

Comment:
In the constructor, allocate this as a string of the same length as the input text, and then assign to each index. At the end, slice it down to the used length, that gets rid of need for the code to be appended character by character to the string.